code/modules/mob/living/carbon/human/ai/action_datums/keep_distance.dm

```diff
@@ -0,0 +1,93 @@
+/datum/ai_action/keep_distance
+	name = "Keep Distance"
+	action_flags = ACTION_USING_LEGS
+
+/datum/ai_action/keep_distance/get_weight(datum/human_ai_brain/brain)
+	var/mob/living/current_target = brain.current_target
+	if(!current_target)
+		return 0
+
+	if(!brain.primary_weapon)
+		return 0
+
+	var/distance = get_dist(brain.tied_human, current_target)
+	var/datum/firearm_appraisal/gun_data = brain.gun_data
+
+	if(current_target.is_mob_incapacitated())
+		if(distance > gun_data.minimum_range)
+			return 10
+
+		if(distance < gun_data.minimum_range)
+			return 10
+
+	else if(brain.in_cover)
+		if(distance < gun_data.minimum_range)
```

Comment:
Yeah, this means we move out of cover only when enemy is closer than the minimum range, E.G. we have a rocket launcher and don't wanna blow ourselves up because hiding behind a cade